cpp/src/binaryop/binaryop.cpp

```diff
@@ -292,21 +292,21 @@ void binary_operation(gdf_column* out, gdf_column* lhs, gdf_column* rhs,
 
 gdf_column binary_operation(const gdf_column& lhs, const gdf_column& rhs,
                             const std::string& ptx, gdf_dtype output_type) {
+  
   CUDF_EXPECTS((lhs.size == rhs.size), "Column sizes don't match");
 
   gdf_column output{};
 
-  // Check for 0 sized data
-  if ((lhs.size == 0) && (rhs.size == 0)) return output;
-  CUDF_EXPECTS((lhs.size == rhs.size), "Column sizes don't match");
-
   if (lhs.valid != nullptr) {
     output = allocate_column(output_type, lhs.size);
   } else if (rhs.valid != nullptr) {
     output = allocate_column(output_type, rhs.size);
   } else {
     output = allocate_column(output_type, lhs.size, false); // don't allocate valid for the output
   }
+  
+  // Check for 0 sized data
+  if (lhs.size == 0) return output;
```

Comment:
This still isn't quite right, because you've allocated memory for the output with `allocate_column`. The returned `output` should be empty.

Reply:
@jrhemstad If `size` is 0 `allocate_column` will just set `dtype` and set `size` to 0 without allocating memory for `data` and `valid`. https://github.com/rapidsai/cudf/blob/83cd4998f9952481cbceca1c4e3ba44f51442b98/cpp/src/column/legacy/column.cpp#L215. Or maybe I understand "empty" incorrectly.

Reply:
Oh, I see what you mean. Yeah, okay, this makes sense then.